### PR TITLE
GSK-2471 GSK-2458 Fixed issue when loading code from uninstalled dep

### PR DESCRIPTION
**Pull request by kevinmessiaen**
## Description

Fixed issue when loading code from uninstalled dep

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [x] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [ ] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [ ] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

**Review by andreybavt — APPROVED** [review]
generally the fix looks good makes sense

File: giskard/core/savable.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def download(cls, uuid: str, client: Optional[GiskardClient], project_key: Optio
 class RegistryArtifact(Artifact[SMT], ABC):
     def _save_locally(self, local_dir: Path):
         with open(Path(local_dir) / "data.pkl", "wb") as f:
-            cloudpickle.dump(self, f, protocol=pickle.DEFAULT_PROTOCOL)
+            dump_by_value(self, f)
```
Comment (andreybavt) [inline giskard/core/savable.py:160]: @kevinmessiaen we should use `dump_by_value` in `CloudpickleSerializableModel.save_model` too, shouldn't we?

Comment (andreybavt) [reply giskard/core/savable.py:160]: probably in `save_model_class` too just to centralize all pickling in one place

**Review by andreybavt — COMMENTED** [review]
Looks good generally, see my comment about centralizing the pickling in one place

File: giskard/core/savable.py (hunk 1 above)
Comment (kevinmessiaen) [reply giskard/core/savable.py:160]: Yes that's right, I'll add this modification

**Pull request merged**